Profiler_Cpp_Test/tests/ConfigTest.cpp

```diff
@@ -0,0 +1,169 @@
+#include <sstream>
+#include "CppUnitTest.h"
+#include "config/Config.h"
+
+using namespace Microsoft::VisualStudio::CppUnitTestFramework;
+
+TEST_CLASS(ConfigTest)
+{
+public:
+
+	TEST_METHOD(EmptyConfigAndNoEnvironment)
+	{
+		Config config = parse(R"()", [](std::string suffix) -> std::string { return "";});
+
+		Assert::AreEqual(size_t(0), config.getProblems().size(), L"number of problems");
+		Assert::AreEqual(true, config.isEnabled(), L"default value should be enabled");
+		Assert::AreEqual(false, config.shouldIgnoreExceptions(), L"default value should be to not ignore exceptions");
+		Assert::AreEqual(size_t(0), config.getEagerness(), L"default value should be no eagerness");
+	}
+
+	TEST_METHOD(EnvironmentMustBeRespected)
+	{
+		Config config = parse(R"()", [](std::string suffix) -> std::string {
+			if (StringUtils::uppercase(suffix) == "IGNORE_EXCEPTIONS") {
+				return "1";
+			}
+			return "";
+		});
+
+		Assert::AreEqual(true, config.shouldIgnoreExceptions(), L"should use value from environment");
+	}
+
+	TEST_METHOD(ConfigFileMustBeRespected)
+	{
+		Config config = parse(R"(
+match:
+  - process: ".*"
+    profiler:
+      ignore_exceptions: true
+)", [](std::string suffix) -> std::string { return ""; });
+
+		Assert::AreEqual(true, config.shouldIgnoreExceptions(), L"should use value from config");
+	}
+
+	TEST_METHOD(OnlyMatchingSectionsAreApplied)
+	{
+		Config config = parse(R"(
+match:
+  - process: ".*doesnt-match"
+    profiler:
+      ignore_exceptions: true
+)", [](std::string suffix) -> std::string { return ""; });
+
+		Assert::AreEqual(false, config.shouldIgnoreExceptions(), L"should be the default value");
+	}
+
+	TEST_METHOD(NoProcessFieldMeansMatchAnyProcess)
+	{
+		Config config = parse(R"(
+match:
+  - profiler:
+      ignore_exceptions: true
+)", [](std::string suffix) -> std::string { return ""; });
+
+		Assert::AreEqual(true, config.shouldIgnoreExceptions(), L"should be the config value");
+	}
+
+	TEST_METHOD(EnvironmentTrumpsConfig)
+	{
+		Config config = parse(R"(
+match:
+  - process: ".*"
+    profiler:
+      targetdir: config
+)", [](std::string suffix) -> std::string {
+			if (StringUtils::uppercase(suffix) == "TARGETDIR") {
+				return "env";
+			}
+			return "";
+		});
+
+		Assert::AreEqual(std::string("env"), config.getTargetDir(), L"should use value from environment");
+	}
+
+	TEST_METHOD(LastMatchWins)
+	{
+		Config config = parse(R"(
+match:
+  - process: ".*"
+    profiler:
+      targetdir: first
+  - process: ".*"
+    profiler:
+      targetdir: last
+)", [](std::string suffix) -> std::string { return ""; });
+
+		Assert::AreEqual(std::string("last"), config.getTargetDir(), L"Should use value from last matching section");
+	}
+
+	TEST_METHOD(MatchingPathSeparators)
+	{
+		Config config = parse(R"(
+match:
+  - process: .*/program.exe
+    profiler:
+      targetdir: forward
+  - process: .*\\program.exe
```

Comment:
for the test it doesn't matter
and in practice this wouldn't matter either since something like "program_exe" being executable on Windows is highly unlikely even and has never happened in any of the setups I've seen so far
it then accidentally being matched even though we only wanted to match program.exe is even more unlikely

Reply:
You did escape the dots in the configuration example, though ;)
I agree it doesn't matter for the test.